scripts/sonic_sku_create.py

```diff
@@ -0,0 +1,789 @@
+                if (answer == "Yes" or answer == "No"):
+                    break
+                else:
+                    print("Valid answers are Yes or No")
+            if (answer == "Yes"):
+                shutil.rmtree(self.new_sku_dir)
+                print("SKU directory: "+ self.new_sku_dir + " was removed")
+            else:
+                print("SKU directory: "+ self.new_sku_dir + " was NOT removed")
+        except OSError as e:
+            print(e.message, file=sys.stderr) 
+
+    def platform_specific(self):
+        # Function that checks for Platform specific restrictions
+        func = self.platform_specific_dict.get(self.platform, lambda: "nothing")
+        return func()
+
+
+    def msn2700_specific(self):
+        # Function that implements the check for platform restrictions of 2700 platform
+        for fp, values in self.fpp_split.items():
+            splt_arr = sorted(values[0])
+            splt = len(splt_arr)
+            try :
+                if ((fp%2) == 1 and splt == 4):
+                    next_fp = fp+1
+                    if (next_fp not in self.fpp_split):
+                        continue
+                    next_fp_idx_arr = sorted(self.fpp_split[next_fp][1])
+                    for i in next_fp_idx_arr :
+                        if (self.verbose):
+                            print("msn2700_specific -> Removing ", self.portconfig_dict[i])
+                        self.portconfig_dict.pop(i)
+                    print("MSN2700 - Front panel port ",next_fp, " should be removed due to port ",fp,"Split by 4")
+                    raise  ValueError()
+                elif ((fp%2) == 0 and splt == 4):
+                    print("MSN2700 -  even front panel ports (",fp,") are not allowed to split by 4")
+                    raise  ValueError()
+            except ValueError:
+                print("Error - Illegal split by 4 ", file=sys.stderr)
+                exit(1)
+
+
+def main(argv):
+    parser = argparse.ArgumentParser(description='Create a new SKU',
+                                                version='1.0.0',
+                                                formatter_class=argparse.RawTextHelpFormatter)
+    group = parser.add_mutually_exclusive_group(required=True)
+    group.add_argument('-f', '--file', action='store', nargs=1, help='SKU definition from xml file. -f OR -m or -j must be provided when creating a new SKU', default=None)
+    group.add_argument('-m', '--minigraph_file', action='store', nargs='?', help='SKU definition from minigraph file. -f OR -m or -j must be provided when creating a new SKU', const="/etc/sonic/minigraph.xml")
+    group.add_argument('-j', '--json_file', action='store', nargs=1, help='SKU definition from config_db.json file. -f OR -m OR -j must be provided when creating a new SKU', default=None)
+    group.add_argument('-pp', '--port_split', action='store', nargs=2, help='port name and split', default=None)
```

Comment:
>-pp [](start = 24, length = 3)

Linux command convention is using single char in short option name. In this case, it's ok to have only long option name.
And one more below.

Reply:
Addressed in next commit.